addons/l10n_uy/tests/test_doc_types.py

```diff
@@ -0,0 +1,60 @@
+from odoo import Command
+from odoo.tests.common import tagged
+
+from odoo.addons.account.tests.common import AccountTestInvoicingCommon
+
+
+@tagged("-at_install", "post_install", "post_install_l10n")
+class TestDocTypes(AccountTestInvoicingCommon):
+
+    @classmethod
+    def setUpClass(cls, chart_template_ref='uy'):
+        super().setUpClass(chart_template_ref=chart_template_ref)
+
+        service_vat_22 = cls.env["product.product"].create({
+            "name": "Virtual Home Staging (VAT 22)",
+            "list_price": 38.25,
+            "standard_price": 45.5,
+            "type": "service",
+            "default_code": "VAT 22",
+        })
+
+        cls.invoice = cls.env["account.move"].with_context(
+            default_move_type="out_invoice").create({
+                "partner_id": cls.env["res.partner"].create({"name": "test partner UY"}).id,
+                "move_type": "out_invoice",
+                "l10n_latam_document_type_id": cls.env.ref("l10n_uy.dc_e_inv_exp").id,
+                "invoice_line_ids": [Command.create({
+                    "product_id": service_vat_22.id,
+                    "quantity": 1.0,
+                    "price_unit": 100.0,
+                    "display_type": "product",
+                })]
+            })
+        cls.invoice.action_post()
+
+    def test_credit_note(self):
+        self.assertEqual(self.invoice.l10n_latam_document_type_id.code, "121", "Not Export e-Invoice")
+
+        refund_wizard = self.env["account.move.reversal"].with_context({
+            "active_ids": [self.invoice.id], "active_model": "account.move"}).create({
+                "reason": "Mercadería defectuosa",
+                "journal_id": self.invoice.journal_id.id
+            })
+        res = refund_wizard.refund_moves()
+        refund = self.env["account.move"].browse(res["res_id"])
+
+        self.assertEqual(refund.l10n_latam_document_type_id.code, "122", "Not Export e-Invoice Credit Note")
+        self.assertEqual(refund.l10n_latam_available_document_type_ids.mapped("code"), ["122"], "Bad Domain")
```

Comment:
this currently fails ?

Reply:
Hi Julien
I hope you are doing well

Yes, the test is failing because is missing the l10n_uy_edi module

1. without the EDI module then the result  of l10n_latam_available_document_type_ids in this line is ["121", "122", "123", "221", "222", "223"]
2. with the edi module install then the result will be 122. The difference is that the EDI module only add to l10n_latam_available_document_type_ids the document types that can be issues, not all the subtype

I understand that is confusing, I will change it to option (1). The reason why I was using (2) was to check both filtering by subtype and can be issue documents.

Reply:
Hi Julien
I hope you are doing well

Yes, the test is failing because is missing the l10n_uy_edi module

1. without the EDI module then the result  of l10n_latam_available_document_type_ids in this line is ["121", "122", "123", "221", "222", "223"]
2. with the edi module install then the result will be 122. The difference is that the EDI module only add to l10n_latam_available_document_type_ids the document types that can be issues, not all the subtype

I understand that is confusing, I will change it to option (1). The reason why I was using (2) was to check both filtering by subtype and can be issue documents.

Same that the case bellow 

@JulienVR

Reply:
ummm the problem is that, since this tests run post_install_l10n, after we have the edi module the test will fail because ill expect ["121"] instead of ["121", "122", "123", "221", "222", "223"]

In this case, it is ok we leave the test as it is and wait to merge it after edi module? or maybe change the post_install_l10n from the test tags?

@JulienVR

Reply:
Isn't the current result `['122', '222']` ?
Anyway, you will have to check if l10n_uy_edi is installed to change the expected value: `self.env['ir.module.module']._get('l10n_uy_edi').state`